include/logging/log_backend.h

```diff
@@ -147,7 +147,7 @@ static inline const struct log_backend *log_backend_get(u32_t idx)
  */
 static inline int log_backend_count_get(void)
 {
-	return ((void *)__log_backends_end - (void *)__log_backends_start) /
+	return ((u8_t *)__log_backends_end - (u8_t *)__log_backends_start) /
```

Comment:
Likely to be a huge bike shed, but personally I'd prefer "char" for the type to use here, which is guaranteed by the standard to be the minimal memory unit size and readable to all.  Our oddball unit types are for unifying size (e.g. long vs. int) and signedness (e.g. char-is-unsigned-on-arm-wtf?).

But others feel probably even more strongly in the other direction, so just a note.

Reply:
*Backseat driving*
What's with the `u8_t`, etc., types in Zephyr btw? Does Zephyr need to support pre-C99 compilers without `<stdint.h>`?

Reply:
It's a mish-mash in the Linux kernel too, but for something like Zephyr it might be reasonable to do a global search-replace I guess, unless there's reasons to avoid the `<stdint.h>` types.

Reply:
We actually did use the standard types pervasively, and the existing types are the **result** of a global search and replace.  :)

I mostly stayed out of the fight, but yeah: inconsistent toolchain support for stdint.h was part of the reasoning.

Reply:
If char-is-unsigned-on-arm-wtf, then I would prefer u8_t cause I know it won't change signedness under my feet -- and it should, knock-on-wood, always be unsigned 8 bit for all supported platforms. Also for consistency.

Reply:
I rather prefer using stdint, but if we use our types, u8_t is better than char, because it explicitly says that is is unsigned. Moreover, this is also a MISRA-C recommendation, if we want be compliant, we need to move in this direction.

Reply:
Leaving as-is.